src/GafferModule/SpreadsheetBinding.cpp

```diff
@@ -117,7 +118,9 @@ class RowsPlugSerialiser : public ValuePlugSerialiser
 
 			// Serialise columns
 
-			if( !reference )
+			IECore::ConstBoolDataPtr columnsNeedSerialisation = Metadata::value<IECore::BoolData>( plug, "spreadsheet:columnsNeedSerialisation" );
+
+			if( !reference && !( columnsNeedSerialisation && !columnsNeedSerialisation->readable() ) )
```

Comment:
The number of negations in this line is getting a bit confusing ... it's not really a problem, but I wonder about flipping the branches around so it could just be:
`if( reference || ( columnsNeedSerialisation && !columnsNeedSerialisation->readable() ) )`

Also, in the branch that does nothing, the comment talks about referenced spreadsheets specifically, but that branch now also applies to the !columnsNeedSerialisation as well.

Reply:
Addressed in 31e9f02 (squashed in), with a separate branch and comment for each of references and metadata.